encoding.go

```diff
@@ -91,11 +93,26 @@ func isDictionaryFormat(encoding format.Encoding) bool {
 	return encoding == format.PlainDictionary || encoding == format.RLEDictionary
 }
 
+func RegisterEncoding(enc encoding.Encoding) bool {
+	ns := encoding.NotSupported{}
+	if enc == ns {
+		return false
+	}
+	if LookupEncoding(enc.Encoding()) != ns {
+		return false
+	}
+	extraEncodings[enc.Encoding()] = enc
+	return true
+}
+
 // LookupEncoding returns the parquet encoding associated with the given code.
 //
 // The function never returns nil. If the encoding is not supported,
 // encoding.NotSupported is returned.
 func LookupEncoding(enc format.Encoding) encoding.Encoding {
+	if enc, ok := extraEncodings[enc]; ok {
+		return enc
+	}
```

Comment:
Could we move this after looking up in the default encodings so it's more explicit that they are taking precedence?